llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1456,6 +1476,107 @@ void AsmPrinter::emitBBAddrMapSection(const MachineFunction &MF) {
         OutContext.reportError(SMLoc(), "Couldn't find the block's index");
       OutStreamer->emitULEB128IntValue(I);
     }
+
+    // Emit call marker table for the Yk JIT.
+    //
+    // The table is a size header, followed by call instruction addresses.
+    //
+    // YKOPT: to save space, instead of using absolute symbol addresses, compute
+    // the distance from the start of the block and use uleb128 encoding.
+    const size_t NumCalls = YkCallMarkerSyms[&MBB].size();
+    OutStreamer->emitULEB128IntValue(NumCalls);
+    for (auto Tup : YkCallMarkerSyms[&MBB]) {
+      // Emit address of the call instruction.
+      OutStreamer->emitSymbolValue(std::get<0>(Tup), getPointerSize());
+      // Emit address of target if known, or 0.
+      MCSymbol *Target = std::get<1>(Tup);
+      if (Target)
+        OutStreamer->emitSymbolValue(Target, getPointerSize());
+      else
+        OutStreamer->emitIntValue(0, getPointerSize());
+    }
+
+    // Emit successor information.
+    //
+    // Each codegenned block gets a record indicating any statically known
+    // successors. The record starts with a `SuccessorKind` byte:
+    enum SuccessorKind {
+      // One successor.
+      Unconditional = 0,
+      // Choice of two successors.
+      Conditional = 1,
+      // A control flow edge known only at runtime, e.g. a return or an indirect
+      // branch.
+      Dynamic = 2,
+    };
+
+    // Then depending of the `SuccessorKind` there is a payload describing the
+    // successors:
+    //
+    // `Unconditional`: [target-address: u64]
+    // `Conditional`:   [taken-address: u64, not-taken-address: u64]
+    // `Dynamic`:       [] (i.e. nothing, becuase nothing is statically known)
+
+    MachineBasicBlock *TBB = nullptr, *FBB = nullptr;
```

Comment:
I think a comment would help here what TBB and FBB stand for.

Reply:
I think it's obvious from the `analyzeBranch()` docs: https://llvm.org/doxygen/classllvm_1_1TargetInstrInfo.html#a0dfb0c744373d4b6112eb343a5b07fc7

Reply:
I didn't have the documentation to hand when I read the code. I assumed there might be an easy description for each, but looking at the docs, there's various combinations with different meanings, so I guess it's fine to leave it as is.

Reply:
Yeah, it's fiddly. To understand this code you really would have to have read the LLVM docs.